lib/pacemaker/pcmk_sched_instances.c

```diff
@@ -414,9 +424,16 @@ check_instance_state(const pe_resource_t *instance, uint32_t *state)
     const GList *iter = NULL;
     uint32_t instance_state = 0; // State of just this instance
 
-    // If the instance has its own children (a cloned group), check each one
-    if (instance->children != NULL) {
-        for (iter = instance->children; iter != NULL; iter = iter->next) {
+    // No need to check further if all conditions have already been detected
```

Comment:
If all bits are set, we can return here before the loop over `instance->children`. If `instance_restarting` is set, we can skip the loop over `instance->actions`, after the block that sets `instance_active` if appropriate.

No worries if you'd rather not have two separate checks.

Reply:
I think it's good enough